src/units/filter.cpp

```diff
@@ -806,6 +806,47 @@ void unit_filter_compound::fill(vconfig cfg)
 					return side_filter(c, args.fc).match(args.u.side());
 				});
 			}
+			else if (child.first == "filter_ability") {
+				create_child(child.second, [](const vconfig& c, const unit_filter_args& args) {
+					for(const config::any_child ab : args.u.abilities().all_children_range()) {
+						if(args.u.ability_matches_filter(ab.cfg, ab.key, c.get_parsed_config())) {
+							return true;
+						}
+					}
+					return false;
+				});
+			}
+			else if (child.first == "filter_ability_active") {
+				create_child(child.second, [](const vconfig& c, const unit_filter_args& args) {
+				assert(display::get_singleton());
```

Comment:
Either return or just skip this test instead of asserting.

Reply:
added return and fix inentation